plugin/magit.vim

```diff
@@ -50,26 +50,25 @@ let g:magit_default_fold_level     = get(g:, 'magit_default_fold_level',
 let g:magit_default_sections       = get(g:, 'magit_default_sections',          ['info', 'global_help', 'commit', 'staged', 'unstaged'])
 let g:magit_discard_untracked_do_delete = get(g:, 'magit_discard_untracked_do_delete',        0)
 
-let g:magit_refresh_gitgutter      = get(g:, 'magit_refresh_gitgutter',         1)
-let g:magit_refresh_vim_signify    = get(g:, 'magit_refresh_vim_signify',       1)
+let g:magit_refresh_gutter         = get(g:, 'magit_refresh_gutter'   ,         1)
+" Should deprecate the following
+let g:magit_refresh_gitgutter      = get(g:, 'magit_refresh_gitgutter',         0)
 let g:magit_warning_max_lines      = get(g:, 'magit_warning_max_lines',         10000)
 
 let g:magit_git_cmd                = get(g:, 'magit_git_cmd'          ,         "git")
 
 execute "nnoremap <silent> " . g:magit_show_magit_mapping . " :call magit#show_magit('v')<cr>"
 
-if ( g:magit_refresh_gitgutter == 1 )
-	autocmd User VimagitUpdateFile
-			\ if ( exists("*gitgutter#process_buffer") ) |
-			\ 	call gitgutter#process_buffer(bufnr(g:magit_last_updated_buffer), 0) |
-			\ endif
-endif
+if ( g:magit_refresh_gutter == 1 || g:magit_refresh_gitgutter == 1)
+
+  if ( exists("*gitgutter#process_buffer") )
```

Comment:
You save a if statement at runtime, but the evaluation is done at plugin initialization. It means that if vimagit is loaded before gitgutter and/or vim-signify, `exists("*gitgutter#process_buffer")` will be evaluated to false.

I do not like either the `if` in the autocmd, but this is the only way to not be dependent on plugin loading order.

Please, only add vim signify support for this PR.

If you find a better way to check gitgutter/vim-signify presence, I'll be happy to merge :)